src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/AbstractAppSchemaMockData.java

```diff
@@ -738,8 +784,30 @@ private String modifyOnlineMappingFileContent(String mappingFileName) throws IOE
                             // copy <parameters> with new db params
                             if (isOracle) {
                                 content.append(AppSchemaTestOracleSetup.DB_PARAMS);
-                            } else {
+                            } else if (isPostgis) {
                                 content.append(AppSchemaTestPostgisSetup.DB_PARAMS);
+                            } else if (isGeoPkg) {
+                                copyGeopkgResourcesFolder();
+                                File[] stations =
+                                        Objects
+                                                .requireNonNull(
+                                                        data.listFiles(
+                                                                (dir, name) ->
+                                                                        name.startsWith(
+                                                                                "appschema")))[0]
+                                                .listFiles(
+                                                        (dir, name) -> name.startsWith("stations"));
```

Comment:
I would directly check for the entire file name with the gpkg extension instead of simply checking the fileName.
Also could you refact a bit this piece of code to make it more readable?

Reply:
done